File: solutions/julia/space-age/1/space-age.jl
orbital_periods = Dict(
    [("mercury", 0.2408467), 
     ("venus", 0.61519726), 
     ("earth", 1.0), 
     ("mars", 1.8808158),
     ("jupiter", 11.862615), 
     ("saturn", 29.447498), 
     ("uranus", 84.016846),
     ("neptune", 164.79132)])

function onEarth(seconds)
    age_onPlanet("earth", seconds)
end

function onMercury(seconds)
    age_onPlanet("mercury", seconds)
end

function onVenus(seconds)
    age_onPlanet("venus", seconds)
end

function onMars(seconds)
     age_onPlanet("mars", seconds)
end

function onJupiter(seconds)
     age_onPlanet("jupiter", seconds)
end

function onSaturn(seconds)
     age_onPlanet("saturn", seconds)
end

function onUranus(seconds)
     age_onPlanet("uranus", seconds)
end

function onNeptune(seconds)
     age_onPlanet("neptune", seconds)
end

function age_onPlanet(planet, seconds)
    earth_year = 31557600
    planet_orbit = orbital_periods[planet]
    round(seconds/(planet_orbit*earth_year); digits=2)
end

File: solutions/kotlin/space-age/1/src/main/kotlin/SpaceAge.kt
class SpaceAge(val seconds : Long) {

   private val relativePeriods = mapOf(
                            "Mercury"   to 0.2408467,
                            "Venus"     to 0.61519726,
                            "Earth"     to	1.0,
                            "Mars"	    to 1.8808158,
                            "Jupiter"	to 11.862615,
                            "Saturn"	to 29.447498,
                            "Uranus"	to 84.016846,
                            "Neptune"	to 164.79132)
   
    private fun ageOn(planet : String) : Double {
        val earthSeconds =  31557600
        return seconds / (relativePeriods.getOrDefault(planet, 1.0) * earthSeconds)
    } 

    fun onEarth(): Double = ageOn("Earth")
    fun onMercury(): Double = ageOn("Mercury")
    fun onVenus(): Double = ageOn("Venus")
    fun onMars(): Double = ageOn("Mars")
    fun onJupiter(): Double = ageOn("Jupiter")
    fun onSaturn(): Double = ageOn("Saturn")
    fun onUranus(): Double = ageOn("Uranus")
    fun onNeptune(): Double = ageOn("Neptune")
}
